### ISSUE TITLE
Mounting single file causes an error after updating to 17.09.0-ce-win33

### ISSUE THREAD

**Issue by unarist** [docker/for-win#1326]
### Expected behavior

Container successfully starts up and mounts specified file from host (Windows).

### Actual behavior

```
R:\>docker run -v R:\dnsmasq.conf:/etc/dnsmasq.conf jpillora/dnsmasq
docker: Error response from daemon: An error occured while sharing drive.
See 'docker run --help'.
```

Also when I run similar container with docker-compose, `docker-compose up` prints below error.

```
ERROR: for dnsmasq  Cannot start service dnsmasq: oci runtime error: container_linux.go:265: starting container process caused "process_linux.go:368: container init caused \"rootfs_linux.go:57: mounting \\\"/R/dnsmasq.conf\\\" to rootfs \\\"/var/lib/docker/overlay2/8d8a5c66b749fc8f99b6815366f489801686c6b31c9324a4e74f3d673ea27a97/merged\\\" at \\\"/var/lib/docker/overlay2/8d8a5c66b749fc8f99b6815366f489801686c6b31c9324a4e74f3d673ea27a97/merged/etc/dnsmasq.conf\\\" caused \\\"not a directory\\\"\""
: Are you trying to mount a directory onto a file (or vice-versa)? Check if the specified host path exists and is the expected type
```

Both command has been worked until 17.06.2-ce-win27 (13194), but not on 17.09.0-ce-win33 (13620). So I've downgraded to 13194.

### Information

  - Diagnostic ID from "Diagnose & Feedback" in the menu.
    2FAE6349-8577-4FB2-A8F9-9F2D03E315CC/2017-11-16_15-33-55
  - a reproducible case if this is a bug, Dockerfiles FTW
  - page URL if this is a docs issue or the name of a man page
  - host distribution and version (Windows version, build number, etc)
     Windows 10 Pro 1703 x64

### Steps to reproduce the behavior

  1. create `C:\dnsmasq.conf` file (you can leave it empty for this reproducion)
  2. run `docker run -v C:\dnsmasq.conf:/etc/dnsmasq.conf jpillora/dnsmasq`

**Comment by unarist** [comment]
Looks like this issue has been fixed at 17.09.1 (or my mistake). Thanks!

**Comment by Megido6** [comment]
same issue with Docker version 17.12.0-ce, build c97c6d6
